lints/lint_san_dns_name_onion_not_ev_cert.go

```diff
@@ -0,0 +1,78 @@
+/*
+ * ZLint Copyright 2019 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package lints
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/util"
+)
+
+const (
+	// onionTLD is a const for the TLD for Tor Hidden Services.
+	onionTLD = ".onion"
+)
+
+type onionNotEV struct{}
+
+// Initialize for an onionNotEV linter is a NOP.
+func (l *onionNotEV) Initialize() error {
+	return nil
+}
+
+// CheckApplies returns true if the certificate is a subscriber certificate that
+// contains a subject name ending in `.onion`.
+func (l *onionNotEV) CheckApplies(c *x509.Certificate) bool {
+	return util.IsSubscriberCert(c) && util.CertificateSubjInTLD(c, onionTLD)
+}
+
+// Execute checks that if the certificate contains any subject names ending in
+// the  .onion TLD that it is also an EV certificate. If there are .onion names
+// and the certificate is not an EV certificate an Error LintResult is returned.
+func (l *onionNotEV) Execute(c *x509.Certificate) *LintResult {
+	isEV := util.IsEV(c.PolicyIdentifiers)
+	names := append(c.DNSNames, c.Subject.CommonName)
+	for _, name := range names {
```

Comment:
This code seems duplicative of what's in CheckApplies. Is there a difference why the loop is explicit here?

Reply:
The loop was duplicated so it could identify the `name` that was present with a `.onion` TLD in the error message.

Looking at it again I don't think it merits the duplication, especially since it would only ever include the _first_ `.onion` subject in the error. I refactored `Execute` in fae6210 to remove the duplicated loop, thanks @zakird !